src/components/TransactionPreview/ActionPreview.tsx

```diff
@@ -818,6 +858,185 @@ const ActionPreview = ({
     );
   }
 
+  if (type === TRANSACTION_BLOCK_TYPE.PLR_DAO_STAKE) {
+    const { fromAsset, fromChainId, toAsset, providerName, providerIconUrl, receiverAddress, hasEnoughPLR } = preview;
+
+    const previewList = crossChainAction?.batchTransactions?.length
+      ? crossChainAction?.batchTransactions.map((action) =>
+          action.type === TRANSACTION_BLOCK_TYPE.ASSET_SWAP ||
+          (!!action.multiCallData && action.type === TRANSACTION_BLOCK_TYPE.SEND_ASSET)
+            ? action.preview
+            : null
+        )
+      : [crossChainAction.preview];
+
+    const fromNetwork = supportedChains.find((supportedChain) => supportedChain.chainId === fromChainId);
+
+    const toNetwork = supportedChains[1];
+
+    const toChainTitle = toNetwork?.title ?? CHAIN_ID_TO_NETWORK_NAME[CHAIN_ID.POLYGON].toUpperCase();
+
+    const fromChainTitle = fromNetwork?.title ?? CHAIN_ID_TO_NETWORK_NAME[fromChainId].toUpperCase();
+
+    const fromAmount = formatAmountDisplay(ethers.utils.formatUnits(fromAsset.amount, fromAsset.decimals));
+    const toAmount = formatAmountDisplay(ethers.utils.formatUnits(toAsset.amount));
+
+    const senderAddress = crossChainAction.useWeb3Provider ? providerAddress : accountAddress;
+
+    return (
+      <Card
+        title="PLR Dao Staking"
+        marginBottom={20}
+        onCloseButtonClick={onRemove}
+        showCloseButton={showCloseButton}
+        additionalTopButtons={additionalTopButtons}
+      >
+        {hasEnoughPLR && (
```

Comment:
You have another check below after this one where you use `!hasEnoughPLR &&` - instead you could shorten this a bit:
```suggestion
        {hasEnoughPLR ? <TrueComponent /> :  <FalseComponent />}
```